File: 242-valid-anagram/242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        char[] ss = s.ToCharArray();
        char[] tt = t.ToCharArray();
        
        Array.Sort(ss);
        Array.Sort(tt);
        
        if(ss.Length != tt.Length) return false;
        
        for(int i = 0; i < tt.Length; i++) {
            if(ss[i] != tt[i]) return false;
        }
        
        return true;
        
    }
}

File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        HashMap<Character, Integer> mp = new HashMap<>();
        if(s.length() != t.length()) {
            return false;
        }
        
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(mp.containsKey(c)) {
                mp.put(c, mp.get(c) + 1);
            }else {
                mp.put(c, 1);
            }
        }
        
        for(int i = 0; i < t.length(); i++) {
            char c = t.charAt(i);
            if(mp.containsKey(c) && mp.get(c) > 0) {
                mp.put(c, mp.get(c) - 1);
            }else {
                return false;
            }
        }
        return true;
    }
}